inMemoryCache.go

```diff
@@ -26,19 +26,19 @@ func NewCache() *InMemoryCache {
 }
 
 // Set sets the `value` at `key` in the cache
-func (c *InMemoryCache) Set(_ context.Context, key interface{}, value Thunk) {
+func (c *InMemoryCache) Set(_ context.Context, key Key, value Thunk) {
 	c.mu.Lock()
-	c.items[key] = value
+	c.items[key.String()] = value
 	c.mu.Unlock()
 }
 
 // Get gets the value at `key` if it exsits, returns value (or nil) and bool
 // indicating of value was found
-func (c *InMemoryCache) Get(_ context.Context, key interface{}) (Thunk, bool) {
+func (c *InMemoryCache) Get(_ context.Context, key Key) (Thunk, bool) {
 	c.mu.RLock()
 	defer c.mu.RUnlock()
 
-	item, found := c.items[key]
+	item, found := c.items[key.String()]
```

Comment:
I noticed these call `key.String()`, but the `Key` interface as written in this commit only specifies methods `Key() string` and `Raw interface{}`.  

`key.String()` reads nicely and plays well with others since that means `Key` implements `Stringer`. Should we make this change?